shared/js/utils.js

```diff
@@ -368,19 +368,24 @@ var inChrome = typeof Components != "undefined" && "utils" in Components;
    * Formats a url for display purposes. This includes converting the
    * domain to punycode, and then decoding the url.
```

Comment:
Let's make this comment clear that these are intended to be used for both display and (uglier in confusing cases) clickthrough, as described by dveditz in comment 12 of the bug, as well as testing the behavior case in the browser.